src/foundry/prosemirror/schema/secret-node.d.mts

```diff
@@ -0,0 +1,27 @@
+import { MarkSpec, Node, NodeSpec } from "prosemirror-model";
+import type SchemaDefinition from "./schema-definition.d.mts";
+import { EditorState, Transaction } from "prosemirror-state";
+
+export default SecretNode;
+
+/**
+ * A class responsible for encapsulating logic around secret nodes in the ProseMirror schema.
+ */
+declare class SecretNode extends SchemaDefinition {
+  static override tag: "section";
+
+  static override get attrs(): Record<string, any>;
+
+  static override getAttrs(el: HTMLElement): boolean | Record<string, any>;
```

Comment:
I sort of noted this already but I suspect this may supposed to be `HTMLSectionElement`; there's one for each element type.

Reply:
For the others (link & image) yes & fixed. Interestingly `section` elements don't have their own type.